src/stepfunctions/steps/utils.py

```diff
@@ -12,6 +12,46 @@
 # permissions and limitations under the License.
 from __future__ import absolute_import
 
+import boto3
+import logging
+
+logger = logging.getLogger('stepfunctions')
+
+
 def tags_dict_to_kv_list(tags_dict):
-    kv_list = [{"Key": k, "Value": v} for k,v in tags_dict.items()]
-    return kv_list
\ No newline at end of file
+    kv_list = [{"Key": k, "Value": v} for k, v in tags_dict.items()]
+    return kv_list
+
+
+"""
+Obtain matching aws partition name based on region
+Return "aws" as default if no region detected
+"""
+def get_aws_partition():
+    partitions = boto3.session.Session().get_available_partitions()
+    cur_region = boto3.session.Session().region_name
+    cur_partition = "aws"
+
+    if cur_region is None:
+        logger.warning("No region detected for the boto3 session. Using default partition: aws")
+        return cur_partition
+
+    for partition in partitions:
+        regions = boto3.session.Session().get_available_regions("stepfunctions", partition)
+        if cur_region in regions:
+            cur_partition = partition
+            return cur_partition
+
+    return cur_partition
+
+
+"""
+ARN builder for task integration
+"""
+def resource_integration_arn_builder(service, api, integration_pattern=None):
```

Comment:
1. This belongs in the service integration module
2. Since this a method, let's name it with a verb instead. e.g. `get/build_service_integration_arn`

Reply:
Makes sense. Will make the change.

Reply:
(2) still applies